### DISCUSSION TITLE
Group Policy question

### DISCUSSION THREAD

**Discussion by jdthird** [Open-Shell/Open-Shell-Menu#1195] [Q&A]
I loved classic shell, and now open shell, for my locked down terminal servers.  The windows group policy options have always been lacking for creating a custom start menu with only the few things users need, and classic / open shell made it SO much easier.

But I'm having an issue with control panel hopefully someone can help with.  I'm doing as always, setting up the menu the way I want it on one system, with the menu style and the items.  I then am going into the registry and pulling the "Menuitems" from the classic menu with one column.  And it works for everything EXCEPT control panel.  Even though the menu items I've copied into the group policy include the things pulled from the registry:

The first line includes the "controlpanelitem" text from the registry: Items=COLUMN_PADDING,COLUMN_BREAK,RecentDocumentsItem,ComputerItem,SettingsMenu,ControlPanelItem,SearchMenu,SEPARATOR,LogOffItem,SearchBoxItem

Down in with all the other lines from the things included from that registry entry are these, specifically for the control panel.
ControlPanelItem.Command=control_panel
ControlPanelItem.Label=$Menu.ControlPanel
ControlPanelItem.Tip=$Menu.ControlPanelTip
ControlPanelItem.Icon=shell32.dll,137
ControlPanelItem.Settings=TRACK_RECENT

When I am in the Control Panel setting of open shell, and have it enabled, under the help section it warns me that I don't have a control panel item in my custom menu so this setting will be ignored.

Clearly I'm missing SOMETHING here.  Maybe another setting that I'm simply overlooking?  Any suggestions would be greatly appreciated.  I can cheat and simply include a shortcut on the start menu itself for the thing I want to make available or even a manual shortcut to the control panel that would open in explorer, but I'd prefer to simply have the control panel visible the normal way with just the couple things we allow.  Plus, I'd just like to know what it is that I'm missing.  That bit of OCD in any admin...

Thanks for any information

**Comment by blackcrack** [comment]
[Classic+Open-Shell_Almanach.odt](https://github.com/Open-Shell/Open-Shell-Menu/files/9916899/Classic%2BOpen-Shell_Almanach.odt)
maybe helps this a bit..
but @ge0rdi could help here too, maybe answer he for you..
and OpenShell can ever need helping hand in programming, if you know anyone who programming, bring he in and help for grow up OpenShell again, because ge0rdi is the only one who unlocks at least the patches and write a minimal code .. if it's really need..

best

**Comment by ge0rdi** [comment]
Hello @jdthird .

I've tried to replicate steps you described, and it just works for me correctly.

Here is what I did:

- Switched to `Classic Style`.
- Removed few items in `Customize Start Menu`, made sure `Control Panel` is there.
- Saved content of `MenuItems1` value in `HKCU\SOFTWARE\OpenShell\StartMenu\Settings`.
- Reverted to clean VM.
- Installed group policy definitions.
- In `Group Policy` -> `Computer Configuration` -> `Administrative Templates` -> `Open-Shell` -> `Open-Shell Menu` -> `Menu items for classic menu with one column`
  - Selected `Enabled`
  - Selected state of policy `Locked to this value`.
  - Pasted previously saved content of `MenuItems1`
- Also forced `Classic one column` style in group policy.
- Installed Open-Shell. 

Open-Shell started right in one column mode with items I configured previously, `Control Panel` was among them.

If this doesn't work for you, I'd suggest trying with clean install.
And eventually describe exact steps you are doing that end up with error.

**Reply by blackcrack** [reply]
bagged in the almanach :blush:

**Reply by jdthird** [reply]
My steps are the exact same thing.  Will not work.  Tried on four different Windows servers, from 2016 through 2022.  

Your clean install suggestion is not a possibility, these are servers in production...

I wonder if something persists from a previous setting saying control panel disabled. Does anything get written to the registry with the other settings hiding the control panel, that may have to be manually cleared out for it to work once that setting is removed?

At least even though it isn't working I can simply add the control panel CPL items individually that I'm allowing to the start menu and THAT works...   I'll be curious on the next RDS server I spin up, if this works out of the gate where nothing else had been applied previously denying the settings.

Thanks anyway.

**Reply by ge0rdi** [reply]
> I wonder if something persists from a previous setting saying control panel disabled. Does anything get written to the registry with the other settings hiding the control panel, that may have to be manually cleared out for it to work once that setting is removed?

You can try to uninstall `Open-Shell` and manually remove:

- `HKEY_LOCAL_MACHINE\SOFTWARE\OpenShell`
- `HKEY_LOCAL_MACHINE\SOFTWARE\Policies\OpenShell`
- `HKEY_CURRENT_USER\SOFTWARE\OpenShell`
- `HKEY_CURRENT_USER\SOFTWARE\Policies\OpenShell`
- `%APPDATA%\OpenShell`
- `%LOCALAPPDATA%\OpenShell`

That are the only places where `Open-Shell` should store something.

**Reply by jdthird** [reply]
Can't delete the default HKLM\software keys since it's a production server and I don't want to mess up anything with logged in users.  Since it's used around the clock with different shifts, it's not an easy thing to do and not worth the effort just for this one thing.  I was able to load the hive for one test user and removed the HKCU keys though, but that by itself didn't help.

I'll simply hope on a clean install where it was never denied previously it will work, and on the ones that I had it on that won't work with these changes, I can at least put the things on the start menu.  I'm achieving what I need so at least they can do what they have to do with my workaround.

Thanks for the info though!

John

**Reply by jdthird** [reply]
Actually I have some time today, I'm going to fire up a test terminal server and see if this works out of the gate with these settings...  At least then I'll know moving forward if I can do this in my environment...

**Reply by jdthird** [reply]
And even on a brand new system, brand new group policies I just created from scratch for this one server, I have the same warning under the control panel setting, that I don't have a control panel item in my custom menu even though it's clearly there...

Something definitely not right with this part, so I'm simply going to ignore trying to get control panel to show and simply create shortcuts for the couple applets I've allowed in the start menu since the rest of it still works just fine, only this one control panel option that once again fails...

Thanks anyway everyone.
